### ISSUE TITLE
Empty reponse in API override with custom response causes panic

### ISSUE THREAD

**Issue by sattvikc** [supertokens/supertokens-golang#107]
For example:

```
originalCreateCodePOST := *originalImplementation.CreateCodePOST
(*originalImplementation.CreateCodePOST) = func(email *string, phoneNumber *string, optionsCreateCode plessmodels.APIOptions, userContextCreateCode supertokens.UserContext) (plessmodels.CreateCodePOSTResponse, error) {
  err := validatePhone(*phoneNumber)
  if err != nil {
    responseJson := map[string]interface{}{
      "message": err.Error(),
    }

    bytes, _ := json.Marshal(responseJson)

    optionsCreateCode.Res.Header().Set("Content-Type", "application/json")
    optionsCreateCode.Res.WriteHeader(400)
    optionsCreateCode.Res.Write(bytes)

    return plessmodels.CreateCodePOSTResponse{}, nil
  }
  return originalCreateCodePOST(email, phoneNumber, optionsCreateCode, userContextCreateCode)
}
return originalImplementation
```

panics inside `createCode.go`

```
if response.OK != nil {
    result = map[string]interface{}{
	    "status":           "OK",
	    "deviceId":         response.OK.DeviceID,
	    "preAuthSessionId": response.OK.PreAuthSessionID,
	    "flowType":         response.OK.FlowType,
    }
} else {
    result = map[string]interface{}{
	    "status":  "GENERAL_ERROR",
	    "message": response.GeneralError.Message,
    }
}
```

trying to access `Message` inside nil (`response.GeneralError`)

**Comment by flixoflax** [comment]
For me, this also happens. But without a custom response. It simply causes a panic when the request is falsy, e.g. no formFields provided...

**Comment by rishabhpoddar** [comment]
This issue has been fixed in golang SDK version v0.7.1
